crypto/fipsmodule/bcm.c

```diff
@@ -196,15 +196,10 @@ BORINGSSL_bcm_power_on_self_test(void) {
   assert_within(rodata_start, kP256Params, rodata_end);
   assert_within(rodata_start, kPKCS1SigPrefixes, rodata_end);
 
-#if defined(OPENSSL_AARCH64) || defined(OPENSSL_ANDROID)
   uint8_t result[SHA256_DIGEST_LENGTH];
   const EVP_MD *const kHashFunction = EVP_sha256();
-#else
-  uint8_t result[SHA512_DIGEST_LENGTH];
-  const EVP_MD *const kHashFunction = EVP_sha512();
-#endif
 
-  static const uint8_t kHMACKey[64] = {0};
+  static const uint8_t kHMACKey[32] = {0};
```

Comment:
magic number

Reply:
This is the desired length of the HMAC key, it can be any value and we choose 32. If the size is larger than the block size of the used hash function then the key will be hashed before use, if it's smaller then it will be padded with zeroes (https://csrc.nist.gov/csrc/media/publications/fips/198/1/final/documents/fips-198-1_final.pdf). But now I'm thinking to change it back to 64 since SHA-256 block size is 64 bytes and not 32 as I mistakenly assumed.